### PR TITLE
Adding SAI PTF Test for LAGHashseedTest

### PR DESCRIPTION
**Pull request by AfraMariyam**

### CODE REVIEW

File: test/saithrift/tests/sail2.py [hunk 1]
```diff
@@ -380,6 +380,175 @@ def runTest(self):
...
+                                            ip_src=ip_src,
+                                            tcp_sport=sport,
+                                            tcp_dport=dport,
+                                            ip_id=109,
+                                            ip_ttl=64)
+
+                send_packet(self, 4, str(pkt))
+                rcv_idx = verify_any_packet_any_port(self, [exp_pkt], [0, 1, 2, 3])
+                count2[rcv_idx] += 1
+                laglist1.append(rcv_idx)
```
Comment (chitra-raghavan) [inline test/saithrift/tests/sail2.py]: The rcv_idx need to be saved in different list (laglist2) for comparing the order of distribution before after changing the hash value.

File: test/saithrift/tests/sail2.py [hunk 2]
```diff
@@ -380,6 +380,175 @@ def runTest(self):
...
+@group('l2')
+@group('lag')
+class LagHashseedTest(sai_base_test.ThriftInterfaceDataPlane):
+    def runTest(self):
+        
+	"""
+        Create a LAG group with 4 ports 1 through 4. Setup static FDB entries for the LAG and send packet to this destination MAC address. 
+        Send 100 packets with varying 5-tuple and check order/sequence of the distribution of packets received on ports 1 through 4. 
+        Change the LAG Hash seed value to 10 and compare the order/sequence of the distribution of packets received for the same set of 100 packets on ports 1 through 4. 
+        Verify that it is different after changing the hash seedimport socket
```
Comment (chitra-raghavan) [inline test/saithrift/tests/sail2.py]: change the description,"import socket"  is already imported in sail2.py

File: test/saithrift/tests/sail2.py [hunk 3]
```diff
@@ -380,6 +380,175 @@ def runTest(self):
...
+                dport += 1
+                 
+            print ("The distribution of packet after changing hash seed:" , count2)
+
+            order_check=0		
+            for i in range(0,max_itrn):
+                if(laglist1[i]==laglist2[i]):
+                    order_check+=1
+                
+	    print ("checking the order/sequence matches in both distribution:" ,order_check)
```
Comment (chitra-raghavan) [inline test/saithrift/tests/sail2.py]: Introduce a condition check to verify the ordering is different before and after changing hash.

Comment (AfraMariyam) [reply test/saithrift/tests/sail2.py]: Updated as per your comments

**Comment by lguohan** [comment]
can you provide the test log?

File: test/saithrift/tests/sail2.py [hunk 4]
```diff
@@ -380,6 +380,175 @@ def runTest(self):
...
+                rcv_idx = verify_any_packet_any_port(self, [exp_pkt], [0, 1, 2, 3])
+                count2[rcv_idx] += 1
+                laglist2.append(rcv_idx)
+                sport += 1
+                dport += 1
+                 
+            print ("The distribution of packet after changing hash seed:" , count2)
+
+            order_check=0		
+            for i in range(0,max_itrn):
```
Comment (MELLANOX-zhichenr) [inline test/saithrift/tests/sail2.py]: 'max_itrn' should be 'max_itrs'

Comment (AfraMariyam) [reply test/saithrift/tests/sail2.py]: Made the changes

**Pull request merged**